### PR TITLE
Make algorithm headings more readable.

### LINKED ISSUE #40: Interface vs internal algorithms
I think it would help the reader / implementer to have a clear distinction between algorithms that belong the interface of MuSig and algorithms which are just internal helper algorithms. We sometimes use a "Internal" suffix but I'm not sure if it's consistent and it's not very explicit.

### PR DESCRIPTION
**Pull request by jonasnick**
Fixes #40

### CODE REVIEW

**Comment by real-or-random** [comment]
I had an extra heading "Specification" here (bikeshedding allowed), and I think it makes sense:
https://github.com/jonasnick/bips/pull/48/files#diff-f6e2a58ec1a76411a03c054e6299de0965e3d2dd34e917af867ddc080b9b4470R188

**Comment by real-or-random** [comment]
Is there any nice way to move the list of inputs below the "Algorithm Foo()" ? Listing them above the algorithm has looked always strange to me.

Attempt:
```
<div>
Algorithm ''KeySort(pk<sub>1..u</sub>)'':
* Inputs:
** The number ''u'' of public keys with ''0 < u < 2^32''
** The public keys ''pk<sub>1..u</sub>'': ''u'' 33-byte arrays
* Return ''pk<sub>1..u</sub>'' sorted in lexicographical order.
</div>
```
The `<div>` suppresses a `<p>` and has better spacing.

File: bip-musig2.mediawiki [hunk 1]
```diff
@@ -225,30 +221,30 @@ The KeyGen Context is a data structure consisting of the following elements:
 
 We write "Let ''(Q, gacc, tacc) = keygen_ctx''" to assign names to the elements of a KeyGen Context.
 
-'''''GetXonlyPubkey(keygen_ctx)''''':
+Algorithm '''GetXonlyPubkey(keygen_ctx)''':
```
Comment (real-or-random) [inline bip-musig2.mediawiki]:
```suggestion
Algorithm ''GetXonlyPubkey(keygen_ctx)'':
```
I think that looks even better and the italics match the font when we call the algorithms. But it's matter of taste in the end.

**Comment by jonasnick** [comment]
> I had an extra heading "Specification"

Not sure why it went away. Added it again.

> Attempt:

Implemented your attempt, looks good to me.

File: bip-musig2.mediawiki (hunk 1 above)
Comment (jonasnick) [reply bip-musig2.mediawiki]: Ok, done.

**Comment by real-or-random** [comment]
> Implemented your attempt, looks good to me.

Yep indeed, I think it's an improvement.

ACK

**Comment by robot-dreams** [comment]
Concept ACK

I agree this seems like an improvement. The updated section organization is cleaner, and the updated algorithm headings avoid the issue of algorithm titles being larger than the sections they're contained in.

Nit: Would it make sense to call them "functions" instead of algorithms, to be consistent with the bullet points in the "Notation" section?

**Comment by jonasnick** [comment]
rebased

**Comment by robot-dreams** [comment]
ACK 681e4a762e037cacd4d994377b0f8594ffc9a840

**Pull request merged**